src/Application.cpp

```diff
@@ -0,0 +1,40 @@
+#include "Application.hpp"
+
+#include <iostream>
+
+namespace ccm {
+	Application::Application(){
+		/* Initialize SDL */
+		if (SDL_Init(SDL_INIT_EVERYTHING) != 0)
+		{
+			std::cerr << "SDL_Init Error: " << SDL_GetError() << std::endl;
+			m_quit = true;
+		}
+
+		/* Create window */
+		m_window = SDL_CreateWindow("Game", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, 1280, 720, SDL_WINDOW_RESIZABLE);
```

Comment:
should we attempt to create a window if we failed to init SDL?

Reply:
Good question. The only other idea I had was to thrown an exception if this failed, but I'm going to just assume no exceptions are wanted in this project?

Reply:
Alternatively we can set m_quit=true and return from the constructor early (which would've been a much better approach than what I submitted)

Reply:
yeah I think the second option would be more ideal. but both are valid options, it is a question if we want exceptions for error handling or not.